src/azure-cli-core/azure/cli/core/_identity.py

```diff
@@ -413,19 +413,25 @@ def save_service_principal_cred(self, sp_entry):
 
     # noinspection PyBroadException
     # pylint: disable=protected-access
-    def add_credential(self, credential):
+    def add_credential(self, credential, **kwargs):
```

Comment:
>add_credential [](start = 8, length = 14)

ADALCredentialCache is used by Azure CLI to store ADAL format token file for backward compatibility. Azure Sphere CLI may not need this. You can add a parameter like `store_ADAL_cache` in Profile and only call add_credential when it is true.

Reply:
Great point, going to do the modification.